src/Core/Migration/V6_7/Migration1742199548MeasurementSystem.php

```diff
@@ -0,0 +1,103 @@
+        CREATE TABLE IF NOT EXISTS `measurement_display_unit` (
+              `id` BINARY(16) NOT NULL,
+              `measurement_system_id` BINARY(16) NOT NULL,
+              `default` TINYINT(1) DEFAULT 0 NOT NULL,
+              `type` VARCHAR(20) NOT NULL,
+              `short_name` VARCHAR(20) NOT NULL,
+              `factor` DOUBLE NOT NULL,
+              `created_at` DATETIME(3) NOT NULL,
+              `updated_at` DATETIME(3) NULL,
+              PRIMARY KEY (`id`),
+              CONSTRAINT `fk.measurement_display_unit.measurement_system_id` FOREIGN KEY (`measurement_system_id`)
+                REFERENCES `measurement_system` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+          ) ENGINE = InnoDB');
+
+        $connection->executeStatement('
+        CREATE TABLE IF NOT EXISTS `measurement_display_unit_translation` (
+            `name` VARCHAR(255) NULL,
+            `plural_name` VARCHAR(255) NULL,
+            `measurement_display_unit_id` BINARY(16) NOT NULL,
+            `language_id` BINARY(16) NOT NULL,
+            `created_at` DATETIME(3) NOT NULL,
+            `updated_at` DATETIME(3) NULL,
+            PRIMARY KEY (`measurement_display_unit_id`,`language_id`),
+            CONSTRAINT `fk.measurement_display_unit_translation.unit_id` FOREIGN KEY (`measurement_display_unit_id`)
+              REFERENCES `measurement_display_unit` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+            CONSTRAINT `fk.measurement_display_unit_translation.language_id` FOREIGN KEY (`language_id`)
+              REFERENCES `language` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+        ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+        ');
+
+        $connection->executeStatement('
+            CREATE TABLE IF NOT EXISTS `product_measurement_display_unit` (
+              `product_id` BINARY(16) NOT NULL,
+              `product_version_id` BINARY(16) NOT NULL,
+              `weight_unit_id` BINARY(16) NOT NULL,
+              `length_unit_id` BINARY(16) NOT NULL,
+              PRIMARY KEY (`product_id`, `product_version_id`, `weight_unit_id`, `length_unit_id`),
+              CONSTRAINT `fk.product_measurement_display_unit.product_id` FOREIGN KEY (`product_id`, `product_version_id`)
+                REFERENCES `product` (`id`, `version_id`) ON DELETE CASCADE ON UPDATE CASCADE,
+              CONSTRAINT `fk.product_measurement_display_unit.weight_unit_id` FOREIGN KEY (`weight_unit_id`)
+                REFERENCES `measurement_display_unit` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+              CONSTRAINT `fk.product_measurement_display_unit.length_unit_id` FOREIGN KEY (`length_unit_id`)
+                REFERENCES `measurement_display_unit` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+            ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+        ');
+
+        if ($this->columnExists($connection, 'sales_channel_domain', 'measurement_system_id')) {
```

Comment:
please add weight_unit_id and length_unit_id for sales_channel_domain
please add default_measurement_system_id, default_weight_unit_id and default_length_unit_id for sales_channel

Reply:
all these fields are not nullable